Tests/RxFeedbackTests/FeedbackLoopsTests.swift

```diff
@@ -0,0 +1,212 @@
+        let system = Observable.system(
+            initialState: "initial",
+            reduce: { oldState, append in
+                return  oldState + append
+            },
+            scheduler: scheduler,
+            scheduledFeedback: feedback
+        )
+        let expected = scheduler.createHotObservable([
+            next(201, "initial")
+            ])
+
+        // Run
+        let results = scheduler.start { system }
+
+        // Test
+        XCTAssertEqual(results.events, expected.recordedEvents)
+    }
+
+    func testNotNilAfterIntialNil() {
+        // Prepare
+        let scheduler = TestScheduler(initialClock: 0)
+        let query: (String) -> Void? = { state in
+            if state == "initial+" {
+                return ()
+            } else {
+                return nil
+            }
+        }
+        let effects: () -> Observable<String> = { .just("_a") }
+        let feedback: (ObservableSchedulerContext<String>) -> Observable<String> = react(query: query, effects: effects)
+        let events = PublishSubject<String>()
+        let system = Observable.system(
+            initialState: "initial",
+            reduce: { oldState, append in
+                return  oldState + append
+            },
+            scheduler: scheduler,
+            scheduledFeedback: feedback, { _ in events.asObservable() }
+        )
+        let expected = scheduler.createHotObservable([
+            next(201, "initial"),
+            next(211, "initial+"),
+            next(213, "initial+_a"),
+            ])
+
+        // Run
+        scheduler.scheduleAt(210) { events.onNext("+") }
+        let results = scheduler.start { system }
+
+        // Test
+        XCTAssertEqual(results.events, expected.recordedEvents)
+    }
+
+    func testIntialNotNil() {
+        // Prepare
+        let scheduler = TestScheduler(initialClock: 0)
+        let query: (String) -> Void? = { state in
+            if state == "initial" {
+                return ()
+            } else {
+                return nil
+            }
+        }
+        let effects: () -> Observable<String> = {
+            return .just("_a")
+        }
+        let feedback: (ObservableSchedulerContext<String>) -> Observable<String> = react(query: query, effects: effects)
+        let system = Observable.system(
+            initialState: "initial",
+            reduce: { oldState, append in
+                return  oldState + append
+            },
+            scheduler: scheduler,
+            scheduledFeedback: feedback
+        )
+        let expected = scheduler.createHotObservable([
+            next(201, "initial"),
+            next(204, "initial_a")
+            ])
+
+        // Run
+        let results = scheduler.start { system }
+
+        // Test
+        XCTAssertEqual(results.events, expected.recordedEvents)
+    }
+
+    func testFeedbacksOrder() {
```

Comment:
I'm not sure what this test is for. `query1` and `query2` are will never simultaneously return a value, so what order is being tested?

Reply:
I expect that if effects 1 and 2 are immediate, then they should have the same order as query 1 and query 2, I mean the same order as they passed into the `system` operator.